src/cascadia/TerminalConnection/TelnetConnection.cpp

```diff
@@ -0,0 +1,296 @@
+                    _reader = Windows::Storage::Streams::DataReader(_socket.InputStream());
+                    _reader.InputStreamOptions(Windows::Storage::Streams::InputStreamOptions::Partial); // returns when 1 or more bytes ready.
+                    _transitionToState(ConnectionState::Connected);
+
+                    // Send newline to bypass User Name prompt.
+                    const auto newline = winrt::to_hstring("\r\n");
+                    WriteInput(newline);
+
+                    // Wait for login.
+                    Sleep(1000);
+
+                    // Send "cls" enter to clear the thing and just look like a prompt.
+                    const auto clearScreen = winrt::to_hstring("cls\r\n");
+                    WriteInput(clearScreen);
+                }
+                catch (...)
+                {
+                    LOG_CAUGHT_EXCEPTION();
+                    _transitionToState(ConnectionState::Failed);
+                }
+            }
+            else if (_isStateOneOf(ConnectionState::Connected))
+            {
+                // Read from socket
+                const auto amountReceived = _socketReceive(_receiveBuffer);
+
+                _session.receive(
+                    telnetpp::bytes{ _receiveBuffer.data(), amountReceived },
+                    [=](telnetpp::bytes data,
+                        std::function<void(telnetpp::bytes)> const& send) {
+                        _applicationReceive(data, send);
+                    },
+                    [=](telnetpp::bytes data) {
+                        _socketSend(data);
+                    });
+            }
+        }
+    }
+    catch (...)
+    {
+        // If the exception was thrown while we were already supposed to be closing, fine. We're closed.
+        // This is because the socket got mad things were being torn down.
+        if (_isStateAtOrBeyond(ConnectionState::Closing))
+        {
+            _transitionToState(ConnectionState::Closed);
+            return S_OK;
+        }
+        else
+        {
+            LOG_CAUGHT_EXCEPTION();
+            _transitionToState(ConnectionState::Failed);
+            return E_FAIL;
+        }
+    }
+
+    // Routine Description:
+    // - Call to buffer up bytes to send to the remote device.
+    // - You must flush before they'll go out.
+    // Arguments:
+    // - data - View of bytes to be sent
+    // Return Value:
+    // - <none>
+    void TelnetConnection::_socketBufferedSend(telnetpp::bytes data)
+    {
+        const uint8_t* first = data.data();
+        const uint8_t* last = data.data() + data.size();
+        const winrt::array_view<const uint8_t> arrayView(first, last);
+        _writer.WriteBytes(arrayView);
+    }
+
+    // Routine Description:
+    // - Flushes any buffered bytes to the underlying socket
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TelnetConnection::_socketFlushBuffer()
+    {
+        _writer.StoreAsync().get();

```

Comment:
You might have to go back to the original implementation where you had a `fire_and_forget` that resumed on a background thread. This, unfortunately, throws an error when you type a character using the keyboard in the Centennial build because the input event is coming in on the STA/on the UI thread.

Reply:
Or just pop the StoreAsync into the ether and never wait for its response

Reply:
OK